.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,28 @@
+<!-- Thank you for contributing a pull request! Here are a few tips to help you:
+
+1. If this is your first contribution, make sure you've read the Contribution Checklist https://developers.mattermost.com/contribute/getting-started/contribution-checklist/
+2. Read our blog post about "Submitting Great PRs" https://developers.mattermost.com/blog/2019-01-24-submitting-great-prs
+3. Take a look at other repository specific documentation at https://developers.mattermost.com/contribute
+-->
+
+#### Summary
+<!--
+A description of what this pull request does.
+-->
+
+#### Ticket Link
+<!--
+If this pull request addresses a Help Wanted ticket, please link the relevant GitHub issue, e.g.
+
+  Fixes https://github.com/mattermost/mattermost-server/issues/XXXXX
+
+Otherwise, link the JIRA ticket.
+-->
+
+#### Related PR's
+<!--
+Your might submit multiple PR's that together fix a ticket. If so, please link them there and remove unrelated fields from the list. Otherwise please remove this section.
```

Comment:
Should this also be in the mattermost-server repository?

```suggestion
List all PRs related to resolving a ticket. For instance, if you submitted a PR to `mattermost/mattermost-server`, please include it here.
```

Reply:
I don't think so. The usual merge order is `server` -> `redux` -> `webapp`. Hence the webapp is the "overview point" of the process. And since github is linking meting back, it's kind of redundant to also link from server to webapp.

Reply:
I see. So, if a PR involves a change to, say, server, redux, mobile, webapp, api-reference and docs, would they link all of them here, but not in the other PRs?

Reply:
Yes.

(Sidenote: since doc changes are done by PM and they have a seperate well working process for this, I would not link it)

Reply:
I see.

What is the objective of having this section in the PR template? Is it to 
- link other work that this PR depends on, so that reviewers know not to merge it before the others have been merged, or
 - is it to link other related PRs to inform reviewers of such work, or
 - something else?

Maybe this helps me better understand a) the value of doing this in the webapp repo, and b) why we would only do this in the webapp repo.

Reply:
1. Yes
2. Yes
3. It a heads up for reviewers who think "oh, this also needs changes in X repo"

Basically it's an "overview" over the process of the ticket.

Reply:
@hanzei Thank you.

If the objective is 1 + 2, and also a heads up for reviewers, then should dependent PRs be linked in Redux as well, for instance? (e.g. server PR). Or mobile?

I'm still not clear why this would only be done in webapp, but not others like mobile.

Reply:
Because GitHub links mentions of PR back, the linking should be a [directed acyclic graph](https://en.wikipedia.org/wiki/Directed_acyclic_graph) of PR dependencies. 

Let's take for example a ticket with changes in server, redux, mobile, webapp and api-reference. The linking would be:
- mobile links to redux and server
- webapp links to redux, server, mobile and api-reference

Please note that the last two link (mobile and api-reference) aren't needed because of dependencies. I just think that there should be an overview over the whole ticket process and the webapp seams to be the right place.

Reply:
> Because GitHub links mentions of PR back, the linking should be a directed acyclic graph of PR dependencies.

But it's not. Redux PR wouldn't be linked in the server PR.

> I just think that there should be an overview over the whole ticket process and the webapp seams to be the right place.

True, but if the change is made only to redux, server and api-reference, then should one of these act as the "overview PR"?

Reply:
> But it's not. Redux PR wouldn't be linked in the server PR.

That's correct. In practice the redux PR doesn't depend on the server PR.

> True, but if the change is made only to redux, server and api-reference, then should one of these act as the "overview PR"?

I've never see this happening. Do you? If this happens, I would like everything in mobile.

Reply:
I still feel it's a confusing experience for contributors not to be consistent, but I don't strongly enough. Also, looks like we've already done so with the existing PR template